daemon/container_unix.go

```diff
@@ -810,6 +812,17 @@ func (container *Container) buildCreateEndpointOptions(n libnetwork.Network) ([]
 		createOptions []libnetwork.EndpointOption
 	)
 
+	if n.Name() == "bridge" || container.NetworkSettings.IsAnonymousEndpoint {
+		createOptions = append(createOptions, libnetwork.CreateOptionAnonymous())
+	}
+
+	// Other configs are applicable only for the endpoint in the network
+	// to which container was connected to on docker run.
+	if n.Name() != container.hostConfig.NetworkMode.NetworkName() &&
+		n.Name() != "bridge" {
```

Comment:
this check essentially gives a free pass for the default `bridge` network even if the container is connected to the default `bridge` network using `docker network connect`.
But, the intention of the check is to make sure we allow the rest of endpoint configs only for the network that is same as the NetworkMode.